magpie/models.py

```diff
@@ -1059,18 +1095,31 @@ class NetworkNode(BaseModel, Base):
 
     id = sa.Column(sa.Integer(), primary_key=True, nullable=False, autoincrement=True)
     name = sa.Column(sa.Unicode(128), nullable=False, unique=True)
-    url = sa.Column(URLType(), nullable=False)
-    users = relationship("User", backref="network_nodes", lazy="dynamic")
+    jwks_url = sa.Column(URLType(), nullable=False)
+    token_url = sa.Column(URLType(), nullable=False)
+    authorization_url = sa.Column(URLType(), nullable=False)
+    redirect_uris = sa.Column(sa.String)
 
-    @property
     def anonymous_user_name(self):
         # type: () -> Str
-        return "{}{}".format(get_constant("MAGPIE_NETWORK_NAME_PREFIX"), self.name)
+        return self.anonymous_user_name_formatter(self.name)
 
-    @property
-    def anonymous_user(self):
-        # type: () -> Optional[User]
-        return self.users.filter(User.user_name == self.anonymous_user_name).first()
+    @staticmethod
+    def anonymous_user_name_formatter(name):
+        # type: (Str) -> Str
+        return "{}{}".format(get_constant("MAGPIE_NETWORK_NAME_PREFIX"), name)
```

Comment:
I just got an idea regarding this to avoid possible conflicts or misinterpretation of user names when reading responses.
How about using explicitly a disallowed character to separate the node prefix from the user name?
Something like `{node-name}:{user-name}`. 
This way, we would never have to worry about misinterpretation or conflicts of network user names.

Reply:
I don't think this is a good idea. 

The user name validation checks still run when we create or update this user so we'd have to explicitly skip those when creating these users. 

That by itself might be ok, but this would also mean that downstream applications (like jupyterhub for example) wouldn't be able to rely on the user names to always be properly formatted and we might run into another problem like the one in https://github.com/bird-house/birdhouse-deploy/pull/396